### ISSUE TITLE
Conventional Observation Operator Configuration

### ISSUE THREAD

**Issue by danholdaway** [GEOS-ESM/swell#149]
[//]: <> (------------------------------------------------------------------)

# Radiosonde

#### Variables

- [x] Winds
- [x] Temperature
- [x] Virtual Temperature
- [x] Surface Pressure
- [x] Moisture

#### UFO Changes
- [x] https://github.com/JCSDA-internal/ufo/pull/2642
- [x] https://github.com/JCSDA-internal/ufo/pull/2666
- [x] https://github.com/JCSDA-internal/ufo/pull/2673

#### Unresolvable differences with GSI
- Duplicate check across multiple observation spaces (https://github.com/JCSDA-internal/ufo/issues/2603) Affecting q, tv and t. Small number of observations (<10) do not match GSI.

[//]: <> (------------------------------------------------------------------)

# Aircraft (no bias correction)

#### Variables

- [x] Winds
- [x] Temperature

#### UFO Changes

- [x] https://github.com/JCSDA-internal/ufo/pull/2673

#### Unresolvable differences with GSI

[//]: <> (------------------------------------------------------------------)

# Satwind

#### Variables

- [x] Winds

#### UFO Changes
- [x] https://github.com/JCSDA-internal/ufo/pull/2642
- [x] https://github.com/JCSDA-internal/ufo/pull/2673

#### Unresolvable differences with GSI
- Count is off by two (out of ~half a million observations)

None

[//]: <> (------------------------------------------------------------------)

# Scatwind

#### Variables

- [x] Winds

#### UFO Changes

- [x] https://github.com/JCSDA-internal/ufo/pull/2642
- [x] https://github.com/JCSDA-internal/ufo/pull/2673

#### Unresolvable differences with GSI

None

[//]: <> (------------------------------------------------------------------)

# Vadwind

#### Variables

- [x] Winds

#### UFO Changes

- [x] https://github.com/JCSDA-internal/ufo/pull/2642
- [x] https://github.com/JCSDA-internal/ufo/pull/2673

#### Unresolvable differences with GSI

None

[//]: <> (------------------------------------------------------------------)

# Surface (land)

#### Variables

- [x] Winds (not used)
- [x] Temperature (not used)
- [x] Virtual Temperature (not used)
- [x] Surface Pressure
- [x] Moisture (not used)

#### UFO Changes

- [x] https://github.com/JCSDA-internal/ufo/pull/2609

#### Unresolvable differences with GSI

- Duplicate check across multiple observation spaces (https://github.com/JCSDA-internal/ufo/issues/2603)

[//]: <> (------------------------------------------------------------------)

# Surface (marine)

#### Variables

- [x] Winds
- [x] Temperature
- [x] Virtual Temperature
- [x] Surface Pressure
- [x] Moisture

#### UFO Changes

- [x] https://github.com/JCSDA-internal/ufo/pull/2609

#### Unresolvable differences with GSI

- Duplicate check across multiple observation spaces (https://github.com/JCSDA-internal/ufo/issues/2603)
- Running with duplicate check results in increasing the temperature errors too much.

Matches to GSI rely on the following changes @gmao-jjin3 made to the GSI:
1. discard observations which have the same location, pressure, and time
2. determine duplicated observations by single precision location an pressure values
3. not to consider co-located air temperature and virtual temperature as duplicated observations.

[//]: <> (------------------------------------------------------------------)

# RASS

#### Variables

- [x] Virtual Temperature (not used)

#### Unresolvable differences with GSI

[//]: <> (------------------------------------------------------------------)

**Comment by danholdaway** [comment]
## 2023-02-08

- @gmao-jjin3: Working on virtual temperature Sondes and surface. Looking into new filters that are needed for infalting error:
- @gmao-wgu: Working on radiosonde winds. Using new GeoVaLs files from Jianjun to assess Pibal.
- @asewnath: Working on ctests for duplicate check filter. Working on new test files. 
- @gmao-msienkie: Building a version of the code to try with aircraft temp. Waiting on Wei's work for the aircraft winds.

**Comment by danholdaway** [comment]
## 2023-02-15

@asewnath
- working on fixes to the duplicate check filter for sonde moisture.
- Pull request with fix to dup check.
- Investigating small differences in the obs error between JEDI and GSI

@gmao-jjin3
- Check UFO duplicate check with ship temperature and virtual temperature
- Check moisture in surface ship.

@gmao-wgu 

- Get code compiled with name changes.
- Work on preparing the code for pull request.

@gmao-msienkie 
- Trying the duplicate check filter with Akira's fix.
- Pulling in radiosonde configuration to test with aircraft.

**Comment by danholdaway** [comment]
## 2023-02-22

@asewnath
- Surface ship moisture

@gmao-jjin3
- Surface marine temperatures

@gmao-wgu 

- Run duplicate check for non-pibal with new dupe check code.
- Look at ctest failures in PR

@gmao-msienkie 
- Run with Wei's branch. 
- Make the PDF plots for aircraft temperature

@gmao-yzhu 
- Refreshing memory on GNNSRO

**Comment by danholdaway** [comment]
## 2023-03-01

@asewnath
- Continuing to investigate surface marine moisture discrepancy. 

@gmao-jjin3
- Testing the duplicate check (in GSI) using difference less than a certain value rather than exactly equal.
- Working on producing aircraft bias correction variables for Meta.

@gmao-wgu 
- Create ufo-data PR and merge in develop
- Check code coverage by putting some prints throughout the filter code and making sure it is exercised.

@gmao-msienkie 
- Looking at 14 temperature observations that do not have duplicate factor applied in UFO.

**Comment by danholdaway** [comment]
## 2023-03-08

@asewnath
- Continue looking at the QC filters for surface marine moisture.

@gmao-jjin3
- Investigate difference between GSI h(x) and UFO h(x) for surface ship moisture.

@gmao-wgu 
- PTO

@gmao-msienkie 
- Run with Wei's branch and uv file. Look at discrepancies that were seen previously.

**Comment by danholdaway** [comment]
## 2023-03-15

@asewnath
- Trying bugfix in ReportedPressureCheck

@gmao-jjin3
- Check GeoVaLs ordering.

@gmao-wgu 
- Looking at GrossCheck PR comments

@gmao-msienkie 
- Review temperature count discrepancy.

**Comment by danholdaway** [comment]
## 2023-03-22

@asewnath
- Keep looking at surface marine

@gmao-jjin3
- Looking at ReportedPressureCheck filter to make sure it works top down.
- Keep looking at surface marine

@gmao-wgu 
- Looking at ReportedPressureCheck filter to make sure it works top down.
- Run radiosonde with flip on read and rewrite filter to make sure it works with the correct top down ordering.

@gmao-msienkie 
- Start to look at how aircraft bias correction might fold into JEDI.

**Comment by asewnath** [comment]
## 2023-03-29

@asewnath
- Keep looking at surface marine

@gmao-jjin3
- Keep looking at surface marine

@gmao-wgu 
- Revisit ASCAT to check NCEP's ambiguity check filter

@gmao-msienkie 
- Quantify sonde adjusted error substitution differences

**Comment by danholdaway** [comment]
## 2023-04-05

@asewnath
- Look at observation values for surface marine and radiosondes.

@gmao-jjin3
- Output GeoVaLs also in pseudo relative humidity.

@gmao-wgu
- Start to look at SSMIS.

@gmao-msienkie
- Help with identifying moisture issues.

**Comment by rgelaro** [comment]
## 2023-04-12
@asewnath

- Look at genqsat code for possible import into UFO (while waiting for new file from Jianjun).

@gmao-jjin3

- Output qsat as GeoVaL, other file changes.

@gmao-wgu

- Work on SSMIS.

@gmao-msienkie

- Examine issues related to sonde release time

**Comment by danholdaway** [comment]
## 2023-04-19

@asewnath
- Look into the few observations that are below the line. Duplicate check test.
- Plot duplicates on a map
- Wind counts too high by 10/7000 in JEDI.

@gmao-jjin3
- Meeting with EMC folks.
- GNSSRO.

@gmao-wgu
- Quality control for SSMIS
- Help Akira with the ship winds

@gmao-msienkie
- GSI ship obs and JEDI kibitzing

**Comment by rgelaro** [comment]
## 2023-04-26

@asewnath 

- compare GSI and UFO sfc marine wind files to reconcile duplicate mismatches

@gmao-jjin3

- Help with sfc marine and bias corr mismatch for MW
- GNSSRO

@gmao-wgu

- QC for SSMIS (bias corr for passive channels mismatch)

@gmao-msienkie

- GSI ship obs and JEDI kibitzing

**Comment by danholdaway** [comment]
## 2023-05-03

@asewnath 

- Probe the ncdiag uv file, ioda uv file and ioda combined file to see if duplicates are being lost in the ioda converter process.

@gmao-jjin3

- Helping with surface marine wind duplicate puzzle.
- GNSSRO

@gmao-wgu

- Rerun GSI with CRTM 2.4
- QC for SSMIS (bias corr for passive channels mismatch)

@gmao-msienkie

- GSI ship obs and JEDI kibitzing

**Comment by danholdaway** [comment]
## 2023-05-10

@asewnath 
- Work on geopotential to geometric height transform in pressure check filter.

@gmao-jjin3

- Work on presentations. x0048 working for GPM

@gmao-wgu

- Help Akira with pressure check filter variable transform changes
- Continuation of SSMIS work.

@gmao-msienkie

- GSI ship obs and JEDI kibitzing

**Comment by danholdaway** [comment]
## 2023-05-24

@danholdaway 

- Work on adding z transform to ufo

@asewnath 

- Working on the pull request with all the changes for moisture and pressure for flip geovals.

@gmao-jjin3

- GNSSRO

@gmao-wgu

- Working on issue with QC part of YAML in variable assignment.

@gmao-msienkie

- GSI ship obs and JEDI kibitzing

**Comment by danholdaway** [comment]
Closing the EPIC. All conventional data matches GSI to the maximum extent possible without replicating undesirable GSI features.
